src/libraries/Pool.sol

```diff
@@ -384,7 +394,7 @@ library Pool {
             // if the protocol fee is on, calculate how much is owed, decrement feeAmount, and increment protocolFee
             if (cache.protocolFee > 0) {
                 // calculate the amount of the fee that should go to the protocol
-                uint256 delta = step.feeAmount * cache.protocolFee / ProtocolFeeLibrary.BIPS_DENOMINATOR;
+                uint256 delta = step.feeAmount * cache.protocolFee / (swapFee + cache.protocolFee);
```

Comment:
I think this is incorrect in the case when swapFee + cache.protocolFee > 100%. I think the denominator should actually be the "effective fee". Or am I misunderstanding this calculation?

Reply:
If I understand correctly, it goes like this.
```
feeAmount = amountIn - amountIn * (1 - swapFee) * (1 - protocolFee) = amountIn * effectiveFee
effectiveFee = swapFee + protocolFee - swapFee * protocolFee
delta = amountIn * (1 - swapFee) * protocolFee = feeAmount * (1 - swapFee) * protocolFee / effectiveFee
```